### DISCUSSION TITLE
[Question]: Can administrator logins using librechat be distinguished from regular users?

### DISCUSSION THREAD

**Discussion by harrisonhxy** [danny-avila/LibreChat#5062] [Q&A]
### What is your question?

Now I have successfully deployed librechat and every time I use it I use it by logging in as an admin user, which is fine!
But now I have a problem: because I have configured the api key for each model in .env, normal users log in with my api key, but they want to use their own api key. this also prevents me from incurring huge costs for calling the model for all the users, especially since I have a lot of students in my group, and I don't want to keep paying me for them to use the api key by me all the time.

So my need now is: is there any way to allow normal users to enter their own api key when logging in, while ensuring normal use by administrators. 

I'm not sure if this functionality has been implemented yet? What I see in the doc documentation so far is that it seems that once the .env file is configured, all users globally are affected.I haven't found anything related to differentiating between administrator and regular user usage.

### More Details

The configuration now is that I use an administrator login, and since the api key is already configured in .env, I can just make the call, which is convenient
<img width="417" alt="image" src="https://github.com/user-attachments/assets/2e84a763-fa18-4bbb-9571-4bdc11675217" />

I want to allow regular users to enter their own api key when logging in. but at the same time keep the convenience of admins logging in directly to use it!

### What is the main subject of your question?

User System/OAuth

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Yes: https://www.librechat.ai/docs/configuration/dotenv#openai

**Comment by harrisonhxy** [comment]
Thanks for noticing this, do you mean setting `OPENAI_API_KEY` to `user_provided`? But with this setup, it seems that the administrator will have to add his own api key even after logging in. my intention is that the administrator logging in will be able to use the .env configured one directly, and the other users will be able to enter their own when they have an api key.

**Reply by danny-avila** [reply]
At the moment, it's either everyone sets their own key or not. Something like you're asking may be possible in the future, but for now this works well and is recommended in your use case.
